Assets/HoloToolkit/Input/Scripts/InputSources/GesturesInput.cs

```diff
@@ -98,12 +98,12 @@ private class SourceData
         {
             public SourceData(InteractionSource interactionSource)
             {
-                SourceId = interactionSource.id;
-                SourceKind = interactionSource.kind;
+                this.interactionSource = interactionSource;
             }
 
-            public readonly uint SourceId;
-            public readonly InteractionSourceKind SourceKind;
+            public uint SourceId { get { return interactionSource.id; } }
+            public InteractionSourceKind SourceKind { get { return interactionSource.kind; } }
```

Comment:
can interactionSource be null?  

and since interactionSource is public, is the casing right?

Reply:
Since this constructor will only be called by Toolkit code (and the InteractionSource comes from the platform to begin with), I don't believe it can ever be null.

Reply:
oh, I see, this isn't a monobehavior.  It has a constructor. :)